### ISSUE TITLE
`:inherit` option for `tool`

### ISSUE THREAD

**Issue by AlexWayfer** [dazuma/toys#46]
Hello.

As we're discussed here: https://github.com/dazuma/toys/issues/8#issuecomment-584336943

I'd like to see (and use) the `:inherit` option, which will inherit all methods, except `run`, I guess.

It'd be handful to rewrite such code:

```ruby
module CommonCode
  private

  def my_method
    # do something
  en
end

tool :some_namespace do
  subtool_apply do
    include CommonCode
  end

  tool :some_tool do
    def run
      my_method
      # something else
    end
  end
end
```

Into this:

```ruby
tool :some_namespace do
  def my_method
    # do something
  end

  tool :some_tool do
    def run
      my_method
      # something else
    end
  end
end
```

**Comment by dazuma** [comment]
I'm still resistant to making this the default behavior because I think it's surprising and I'm worried about unintended effects. However, I'm thinking of creating a configuration mechanism (#52) that would enable tool-scoped customization of the Toys behavior. Then we could activate this behavior for a particular tool subtree by setting a configuration:

```ruby
tool :some_namespace do
  config.tools.propagate_helper_methods = true

  # This helper method is available in all subtools, recursively
  def my_method
    # do something
  end

  tool :some_tool do
    def run
      my_method # This works
    end
  end
end

tool :another_namespace do
  # This helper does not propagate to subtools because
  # propagate_helper_methods is false by default
  def my_method
    # do something
  end

  tool :some_tool do
    def run
      my_method # NoMethodError
    end
  end
end
```

**Comment by AlexWayfer** [comment]
> I'm still resistant to making this the default behavior because I think it's surprising and I'm worried about unintended effects. However, I'm thinking of creating a configuration mechanism (#52) that would enable tool-scoped customization of the Toys behavior. Then we could activate this behavior for a particular tool subtree by setting a configuration:

OK, thank you, I'm OK with this idea.

**Comment by dazuma** [comment]
This has been implemented and will be released in version 0.12. The final syntax is:

```ruby
tool :some_namespace do
  settings.propagate_helper_methods = true

  # This helper method is available in all subtools, recursively
  def my_method
    # do something
  end

  tool :some_tool do
    def run
      my_method # This works
    end
  end
end
```

It is also possible to do this using the new class-based tool definition syntax, also to be released in version 0.12.

```ruby
class Test < Toys::Tool
  def common_test_method
    # do something
  end

  class Unit < Test
    def run
      common_test_method
    end
  end
end

# Now you can `toys test unit`
```
